near-rs/contract/src/lib.rs

```diff
@@ -2,31 +2,36 @@ extern crate near_sdk;
 use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
 use near_sdk::collections::LookupMap;
 use near_sdk::serde::Serialize;
-use near_sdk::{env, near_bindgen, require, AccountId, EpochHeight};
+use near_sdk::serde_json::to_string;
+use near_sdk::{env, near_bindgen, require, EpochHeight, PublicKey};
 
 #[near_bindgen]
 #[derive(BorshDeserialize, BorshSerialize)]
 pub struct FractalRegistry {
     pub grants_by_id: LookupMap<String, Grant>,
 
-    pub grant_ids_by_owner: LookupMap<AccountId, Vec<String>>,
-    pub grant_ids_by_grantee: LookupMap<AccountId, Vec<String>>,
+    pub grant_ids_by_owner: LookupMap<PublicKey, Vec<String>>,
+    pub grant_ids_by_grantee: LookupMap<PublicKey, Vec<String>>,
     pub grant_ids_by_data_id: LookupMap<String, Vec<String>>,
 }
 
 #[derive(BorshDeserialize, BorshSerialize, Serialize, Clone)]
 #[serde(crate = "near_sdk::serde")]
 pub struct Grant {
-    owner: AccountId,
-    grantee: AccountId,
+    owner: PublicKey,
+    grantee: PublicKey,
     data_id: String,
     locked_until: EpochHeight,
 }
 
 pub fn derive_grant_id(grant: &Grant) -> String {
     let id = format!(
         "{}{}{}{}",
-        grant.owner, grant.grantee, grant.data_id, grant.locked_until,
+        // FIXME: What's the right way to serialize this without the '"'s?
+        to_string(&grant.owner).unwrap().replace("\"", ""),
```

Comment:
@juliosantos have you perhaps learned how to work with serde? I'd like to get rid of this ugly hack.

`PublicKey` is not `Display`. It is `Debug`, but it feels wrong to rely on that.

Reply:
Júlio fixed this with his own commits.